app/exerlog/lib/UI/calendar/widgets/logout_button.dart

```diff
@@ -24,7 +25,46 @@ class _LogoutButtonState extends State<LogoutButton> {
             setState(() {
               _isLoading = true;
             });
-            await FirebaseAuth.instance.signOut();
+            bool shouldLogout = false;
+            await showDialog(
+              context: context,
+              builder: (context) {
+                return AlertDialog(
+                  backgroundColor: backgroundColor,
+                  title: Text(
+                    "Logout",
+                    style: TextStyle(
+                      color: mediumTitleStyleWhite.color,
+                    ),
+                  ),
+                  content: Text(
+                    'Are you sure you want to log out?',
+                    style: TextStyle(
+                      color: smallTitleStyleWhite.color,
+                    ),
+                  ),
+                  actions: [
+                    TextButton(
+                      child: Text('Cancel'),
+                      onPressed: (() {
```

Comment:
Seeing as both of the onPressed functions largely do the same thing, the only difference being the value of shouldLogout, you can make a function for this.

```
void _onButtonPressed(bool shouldLogout){
 shouldLogout = shouldLogout;
 Navigator.pop(context)
}
```
and then 
```
onPressed: ()=> _onButtonPressed(true or false)
```

Reply:
If shouldLogout is true, do firebase.logout, else do nothing... 
Better put that firebase.logout inside of confirm button. Keep it simple. 10 lines less